[SPARK-28527][SQL][TEST] Re-run all the tests in SQLQueryTestSuite via Thrift Server

wangyum opened the pull request:
### What changes were proposed in this pull request?
This PR build a test framework that directly re-run all the tests in `SQLQueryTestSuite` via Thrift Server. But it's a little different from `SQLQueryTestSuite`:
1. Can not support [UDF testing](https://github.com/apache/spark/blob/44e607e9213bdceab970606fb15292db2fe157c2/sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala#L293-L297).
2. Can not support `DESC` command and `SHOW` command because `SQLQueryTestSuite` [formatted the output](https://github.com/apache/spark/blob/1882912cca4921d3d8c8632b3bb34e69e8119791/sql/core/src/main/scala/org/apache/spark/sql/execution/HiveResult.scala#L38-L50.).

When building this framework, found two bug:
[SPARK-28624](https://issues.apache.org/jira/browse/SPARK-28624): `make_date` is inconsistent when reading from table
[SPARK-28611](https://issues.apache.org/jira/browse/SPARK-28611): Histogram's height is different

found two features that ThriftServer can not support:
[SPARK-28636](https://issues.apache.org/jira/browse/SPARK-28636): ThriftServer can not support decimal type with negative scale
[SPARK-28637](https://issues.apache.org/jira/browse/SPARK-28637): ThriftServer can not support interval type

Also, found two inconsistent behavior:
[SPARK-28620](https://issues.apache.org/jira/browse/SPARK-28620): Double type returned for float type in Beeline/JDBC
[SPARK-28619](https://issues.apache.org/jira/browse/SPARK-28619):  The golden result file is different when tested by `bin/spark-sql`


### Why are the changes needed?

Improve the overall test coverage for Thrift Server. 


### Does this PR introduce any user-facing change?
No


### How was this patch tested?
N/A

wangyum committed:
Directly re-run all the tests in SQLQueryTestSuite via Thrift Server

wangyum committed:
avoid test with ConfigSets

wangyum committed:
Merge remote-tracking branch 'upstream/master' into SPARK-28527

wangyum committed:
Add ThriftServerQueryTestSuite to RunInTheirOwnDedicatedJvm

wangyum committed:
Move startThriftServer from beforeEach to beforeAll

wangyum committed:
Merge remote-tracking branch 'upstream/master' into SPARK-28527

wangyum committed:
Maven can not get the correct baseResourcePath

wangyum committed:
Fix pgSQL/text.sql

wangyum committed:
Workaround Maven can't get correct resource directory when resources in other jars.

wangyum commented:
Please note that this PR need to `test-hadoop3.2` and `test-maven`.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/14688/
Test PASSed.

SparkQA commented:
**[Test build #109639 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109639/testReport)** for PR 25567 at commit [`beafc57`](https://github.com/apache/spark/commit/beafc578be109690d23cf03d12e56b924241e277).

SparkQA commented:
**[Test build #109639 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109639/testReport)** for PR 25567 at commit [`beafc57`](https://github.com/apache/spark/commit/beafc578be109690d23cf03d12e56b924241e277).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/109639/
Test PASSed.

wangyum commented:
retest this please

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/14690/
Test PASSed.

SparkQA commented:
**[Test build #109641 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109641/testReport)** for PR 25567 at commit [`beafc57`](https://github.com/apache/spark/commit/beafc578be109690d23cf03d12e56b924241e277).

wangyum commented:
sbt with hadoop-2.7 test results:
https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109639/testReport/org.apache.spark.sql/SQLQueryTestSuite
https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109639/testReport/org.apache.spark.sql.hive.thriftserver/ThriftServerQueryTestSuite

SparkQA commented:
**[Test build #109641 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109641/testReport)** for PR 25567 at commit [`beafc57`](https://github.com/apache/spark/commit/beafc578be109690d23cf03d12e56b924241e277).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/109641/
Test PASSed.

wangyum commented:
retest this please

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/14706/
Test PASSed.

SparkQA commented:
**[Test build #109657 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109657/testReport)** for PR 25567 at commit [`beafc57`](https://github.com/apache/spark/commit/beafc578be109690d23cf03d12e56b924241e277).

dongjoon-hyun commented:
So, only `test-hadoop3.2` and `test-mave` testing is remaining in this PR? For now, it looks working.

SparkQA commented:
**[Test build #109657 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109657/testReport)** for PR 25567 at commit [`beafc57`](https://github.com/apache/spark/commit/beafc578be109690d23cf03d12e56b924241e277).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/109657/
Test FAILed.

wangyum commented:
Yes. It works. But `pgSQL/text.sql` failed:
```
- pgSQL/text.sql *** FAILED ***
  "-1		
  -2		
  -3		
  -4		
  -5		
  0		
  1	a	j
  2	ah	oj
  3	aho	hoj
  4	ahoj	ahoj
  5	ahoj	ahoj" did not contain "Exception" Exception did not match for query #19
  select i, left('ahoj', i), right('ahoj', i) from range(-5, 6) t(i) order by i, expected: -1		
  -2		
  -3		
  -4		
  -5		
  0		
  1	a	j
  2	ah	oj
  3	aho	hoj
  4	ahoj	ahoj
  5	ahoj	ahoj, but got: java.sql.SQLException
  org.apache.spark.sql.catalyst.parser.ParseException: 
  no viable alternative at input 'left'(line 1, pos 10)
  
  == SQL ==
  select i, left('ahoj', i), right('ahoj', i) from range(-5, 6) t(i) order by i
  ----------^^^ (ThriftServerQueryTestSuite.scala:178)
```

wangyum committed:
Fix no viable alternative:
  org.apache.spark.sql.catalyst.parser.ParseException:
  no viable alternative at input 'left'(line 1, pos 10)

  == SQL ==
  select i, left('ahoj', i), right('ahoj', i) from range(-5, 6) t(i) order by i
  ----------^^^

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/14714/
Test PASSed.

SparkQA commented:
**[Test build #109665 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109665/testReport)** for PR 25567 at commit [`4aa5ee2`](https://github.com/apache/spark/commit/4aa5ee26bf8db14326fcf6a437ef08dbb7deed87).

SparkQA commented:
**[Test build #109665 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109665/testReport)** for PR 25567 at commit [`4aa5ee2`](https://github.com/apache/spark/commit/4aa5ee26bf8db14326fcf6a437ef08dbb7deed87).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/109665/
Test PASSed.

wangyum commented:
retest this please

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/14715/
Test PASSed.

SparkQA commented:
**[Test build #109666 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109666/testReport)** for PR 25567 at commit [`4aa5ee2`](https://github.com/apache/spark/commit/4aa5ee26bf8db14326fcf6a437ef08dbb7deed87).

SparkQA commented:
**[Test build #109666 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109666/testReport)** for PR 25567 at commit [`4aa5ee2`](https://github.com/apache/spark/commit/4aa5ee26bf8db14326fcf6a437ef08dbb7deed87).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/109666/
Test PASSed.

wangyum commented:
retest this please

SparkQA commented:
**[Test build #109672 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109672/testReport)** for PR 25567 at commit [`4aa5ee2`](https://github.com/apache/spark/commit/4aa5ee26bf8db14326fcf6a437ef08dbb7deed87).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/14722/
Test PASSed.

SparkQA commented:
**[Test build #109672 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109672/testReport)** for PR 25567 at commit [`4aa5ee2`](https://github.com/apache/spark/commit/4aa5ee26bf8db14326fcf6a437ef08dbb7deed87).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/109672/
Test PASSed.

HyukjinKwon commented:
Oh! @wangyum sorry I didn't know you opened a PR here.

HyukjinKwon commented:
I opened a followup in your branch https://github.com/wangyum/spark/pull/27. I think using Spark home is more consistent in Spark code base, and more robust.

HyukjinKwon reviewed (approved):
Looks good except wangyum#27

HyukjinKwon committed:
Use Spark home when it detects 'sql-tests' directory (#27)

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/14737/
Test PASSed.

SparkQA commented:
**[Test build #109689 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109689/testReport)** for PR 25567 at commit [`2254809`](https://github.com/apache/spark/commit/2254809a4de19030ecf3b5561aafaeb5e1557b4b).

HyukjinKwon commented:
retest this please

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/14738/
Test PASSed.

HyukjinKwon commented:
retest this please

SparkQA commented:
**[Test build #109690 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109690/testReport)** for PR 25567 at commit [`2254809`](https://github.com/apache/spark/commit/2254809a4de19030ecf3b5561aafaeb5e1557b4b).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/14739/
Test PASSed.

HyukjinKwon commented:
retest this please

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/14740/
Test PASSed.

SparkQA commented:
**[Test build #109691 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109691/testReport)** for PR 25567 at commit [`2254809`](https://github.com/apache/spark/commit/2254809a4de19030ecf3b5561aafaeb5e1557b4b).

SparkQA commented:
**[Test build #109689 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109689/testReport)** for PR 25567 at commit [`2254809`](https://github.com/apache/spark/commit/2254809a4de19030ecf3b5561aafaeb5e1557b4b).
 * This patch **fails due to an unknown error code, -9**.
 * This patch merges cleanly.
 * This patch adds no public classes.

SparkQA commented:
**[Test build #109691 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109691/testReport)** for PR 25567 at commit [`2254809`](https://github.com/apache/spark/commit/2254809a4de19030ecf3b5561aafaeb5e1557b4b).
 * This patch **fails due to an unknown error code, -9**.
 * This patch merges cleanly.
 * This patch adds no public classes.

SparkQA commented:
**[Test build #109690 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109690/testReport)** for PR 25567 at commit [`2254809`](https://github.com/apache/spark/commit/2254809a4de19030ecf3b5561aafaeb5e1557b4b).
 * This patch **fails due to an unknown error code, -9**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/109689/
Test FAILed.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/109691/
Test FAILed.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/109690/
Test FAILed.

HyukjinKwon commented:
retest this please

SparkQA commented:
**[Test build #109693 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109693/testReport)** for PR 25567 at commit [`2254809`](https://github.com/apache/spark/commit/2254809a4de19030ecf3b5561aafaeb5e1557b4b).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/14743/
Test PASSed.

HyukjinKwon commented:
retest this please

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/14744/
Test PASSed.

SparkQA commented:
**[Test build #109694 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109694/testReport)** for PR 25567 at commit [`2254809`](https://github.com/apache/spark/commit/2254809a4de19030ecf3b5561aafaeb5e1557b4b).

HyukjinKwon commented:
retest this please

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/14745/
Test PASSed.

SparkQA commented:
**[Test build #109695 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109695/testReport)** for PR 25567 at commit [`2254809`](https://github.com/apache/spark/commit/2254809a4de19030ecf3b5561aafaeb5e1557b4b).

SparkQA commented:
**[Test build #109694 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109694/testReport)** for PR 25567 at commit [`2254809`](https://github.com/apache/spark/commit/2254809a4de19030ecf3b5561aafaeb5e1557b4b).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/109694/
Test PASSed.

SparkQA commented:
**[Test build #109693 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109693/testReport)** for PR 25567 at commit [`2254809`](https://github.com/apache/spark/commit/2254809a4de19030ecf3b5561aafaeb5e1557b4b).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/109693/
Test PASSed.

HyukjinKwon commented:
retest this please

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/14752/
Test PASSed.

SparkQA commented:
**[Test build #109702 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109702/testReport)** for PR 25567 at commit [`2254809`](https://github.com/apache/spark/commit/2254809a4de19030ecf3b5561aafaeb5e1557b4b).

SparkQA commented:
**[Test build #109695 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109695/testReport)** for PR 25567 at commit [`2254809`](https://github.com/apache/spark/commit/2254809a4de19030ecf3b5561aafaeb5e1557b4b).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/109695/
Test PASSed.

SparkQA commented:
**[Test build #109702 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109702/testReport)** for PR 25567 at commit [`2254809`](https://github.com/apache/spark/commit/2254809a4de19030ecf3b5561aafaeb5e1557b4b).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/109702/
Test FAILed.

HyukjinKwon commented:
retest this please

SparkQA commented:
**[Test build #109710 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109710/testReport)** for PR 25567 at commit [`2254809`](https://github.com/apache/spark/commit/2254809a4de19030ecf3b5561aafaeb5e1557b4b).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/14760/
Test PASSed.

SparkQA commented:
**[Test build #109710 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109710/testReport)** for PR 25567 at commit [`2254809`](https://github.com/apache/spark/commit/2254809a4de19030ecf3b5561aafaeb5e1557b4b).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/109710/
Test FAILed.

wangyum commented:
```
19/08/25 22:03:34 INFO ThriftCLIService: Starting ThriftBinaryCLIService on port 13470 with 5...500 worker threads
- subquery/exists-subquery/exists-basic.sql *** FAILED ***
  java.sql.SQLException: Could not open client transport with JDBC Uri: jdbc:hive2://localhost:13470: java.net.ConnectException: Connection refused (Connection refused)
```

wangyum commented:
retest this please

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/14764/
Test PASSed.

SparkQA commented:
**[Test build #109715 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109715/testReport)** for PR 25567 at commit [`2254809`](https://github.com/apache/spark/commit/2254809a4de19030ecf3b5561aafaeb5e1557b4b).

SparkQA commented:
**[Test build #109715 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109715/testReport)** for PR 25567 at commit [`2254809`](https://github.com/apache/spark/commit/2254809a4de19030ecf3b5561aafaeb5e1557b4b).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/109715/
Test FAILed.

wangyum commented:
```
CoarseGrainedSchedulerBackendSuite:
- serialized task larger than max RPC message size
- compute max number of concurrent tasks can be launched
- compute max number of concurrent tasks can be launched when spark.task.cpus > 1
- compute max number of concurrent tasks can be launched when some executors are busy
- custom log url for Spark UI is applied
- extra resources from executor *** FAILED ***
  The code passed to eventually never returned normally. Attempted 322 times over 5.0078623669999995 seconds. Last failure message: ArrayBuffer("1", "3") did not equal Array("0", "1", "3"). (CoarseGrainedSchedulerBackendSuite.scala:249)
```

wangyum commented:
retest this please

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/14767/
Test PASSed.

SparkQA commented:
**[Test build #109718 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109718/testReport)** for PR 25567 at commit [`2254809`](https://github.com/apache/spark/commit/2254809a4de19030ecf3b5561aafaeb5e1557b4b).

SparkQA commented:
**[Test build #109718 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109718/testReport)** for PR 25567 at commit [`2254809`](https://github.com/apache/spark/commit/2254809a4de19030ecf3b5561aafaeb5e1557b4b).
 * This patch **fails due to an unknown error code, -9**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/109718/
Test FAILed.

wangyum commented:
retest this please

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/14773/
Test PASSed.

SparkQA commented:
**[Test build #109722 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109722/testReport)** for PR 25567 at commit [`2254809`](https://github.com/apache/spark/commit/2254809a4de19030ecf3b5561aafaeb5e1557b4b).

SparkQA commented:
**[Test build #109722 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109722/testReport)** for PR 25567 at commit [`2254809`](https://github.com/apache/spark/commit/2254809a4de19030ecf3b5561aafaeb5e1557b4b).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/109722/
Test PASSed.

HyukjinKwon commented:
Merged to master.

wangyum commented:
Could we ignore this test on jenkins?

wangyum commented:
I plan to ignore this test: https://github.com/apache/spark/pull/25592 We can run it manually.

HyukjinKwon commented:
I think we can ignore them for now. Looks @wangyum going to fix the flakiness.

HyukjinKwon commented:
I am going to merge #25592. If we prefer to revert them, we can revert both - I don't mind.

juliuszsompolski commented on sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/ThriftServerQueryTestSuite.scala:
```diff
@@ -0,0 +1,358 @@
+    (1 to 3).foldLeft(Try(startThriftServer(listeningPort, 0))) { case (started, attempt) =>
+      started.orElse {
+        listeningPort += 1
+        Try(startThriftServer(listeningPort, attempt))
```
@wangyum 
I think this is the cause of flakiness.

This goes down to https://github.com/apache/spark/blob/master/sql/hive-thriftserver/v1.2.1/src/main/java/org/apache/hive/service/cli/thrift/ThriftCLIService.java#L178, which starts the service in an asynchronous background thread. It can race and try to connect in the test before the server actually gets initialized.
Moving it to `beforeAll()` and adding a few seconds sleep should help.
Or we could transplant the whole way from `HiveThriftServer2Test` that starts uses the start-thriftserver.sh script and monitors it's log until its started. Would it be possible to make `HiveThriftServer2Test` a trait that can be mixed in here?

wangyum replied:
Move it to `beforeAll()` also unstable: https://github.com/apache/spark/pull/25567#issuecomment-524507309

Anyway, I will try to move it to `beforeAll()` again.

MaxGekk commented on sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/ThriftServerQueryTestSuite.scala:
```diff
@@ -0,0 +1,358 @@
+ *   2. Support DESC command.
+ *   3. Support SHOW command.
+ */
+class ThriftServerQueryTestSuite extends SQLQueryTestSuite {
```
I am trying to troubleshoot an jenkins build failure on this this, but the regular command which I use to run `SQLQueryTestSuite` 
```
SPARK_GENERATE_GOLDEN_FILES=1 build/sbt "sql/test-only *ThriftServerQueryTestSuite -- -z interval.sql"
```
Could someone help me to run `ThriftServerQueryTestSuite`?

HyukjinKwon replied:
@wangyum, can you write some guides in the documentation about how we run each tests?

wangyum replied:
https://github.com/apache/spark/pull/26587
